core/src/main/java/pl/agh/edu/utils/RandomUtils.java

```diff
@@ -0,0 +1,114 @@
+package pl.agh.edu.utils;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalUnit;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+
+import pl.agh.edu.model.time.Time;
+
+public class RandomUtils {
+
+	public static int randomInt() {
+		return ThreadLocalRandom.current().nextInt();
+	}
+
+	public static int randomInt(int bound) {
+		return ThreadLocalRandom.current().nextInt(bound);
+	}
+
+	public static int randomInt(int origin, int bound) {
+		return ThreadLocalRandom.current().nextInt(origin, bound);
+	}
+
+	public static double randomDouble() {
+		return ThreadLocalRandom.current().nextDouble();
+	}
+
+	public static double randomDouble(double bound) {
+		return ThreadLocalRandom.current().nextDouble(bound);
+	}
+
+	public static double randomDouble(double origin, double bound) {
+		return ThreadLocalRandom.current().nextDouble(origin, bound);
+	}
+
+	public static double randomGaussian(double mean, double standardDeviation) {
+		return ThreadLocalRandom.current().nextGaussian() * standardDeviation + mean;
+	}
+
+	public static boolean randomBooleanWithProbability(double probability) {
+		if (probability < 0 || probability > 1) {
+			throw new IllegalArgumentException("Probability must be between 0 and 1.");
+		}
+		return ThreadLocalRandom.current().nextDouble() < probability;
+	}
+
+	public static <T> T randomArrayElement(T[] array) {
+		return array[ThreadLocalRandom.current().nextInt(array.length)];
+	}
+
+	public static <T extends Enum<T>> T randomEnumElement(Class<T> enumClass) {
+		return randomArrayElement(enumClass.getEnumConstants());
+	}
+
+	public static <T> T randomListElement(List<T> list) {
+		return list.get(ThreadLocalRandom.current().nextInt(0, list.size()));
+	}
+
+	public static LocalDate randomDate(LocalDate origin, LocalDate bound) {
+		if (!origin.isBefore(bound)) {
```

Comment:
mozna extractowac jako osobną metode to samo jest w metodzierandom local time

Reply:
wydaje mi się, że to by pogorszyło czytelność kodu + nie zbyt mie się podoba taka metoda

Reply:
jednak się nie da wyciągnąć do osobnej metody bo metoda ta musiała by działać na LocalDateTime, LocalTime, LocalDate a nie implementują one wspólnego interfejsu z metodą isBefore